buildSrc/src/main/java/quilt/internal/plugin/MapV2Plugin.java

```diff
@@ -0,0 +1,261 @@
+ *          {@link JavaExec#classpath(Object...) classpath} to include the
+ *          {@value UNPICK_CLI_CONFIGURATION_NAME} {@link Configuration}
+ *    <li> {@linkplain TaskCollection#configureEach(Action) configures}
+ *          the following defaults for {@link MappingsV2JarTask}s:
+ *          <ul>
+ *              <li> {@link MappingsV2JarTask#getUnpickMeta() unpickMeta}: {@link QuiltMappingsExtension}'s
+ *                   {@link QuiltMappingsExtension#getUnpickMeta() unpickMeta}
+ *              <li> {@link MappingsV2JarTask#getUnpickDefinition() unpickDefinition}:
+ *                   {@link CombineUnpickDefinitionsTask}'s {@link CombineUnpickDefinitionsTask#getOutput() output}
+ *          </ul>
+ * </ul>
+ *
+ * Note: v2 {@value Constants#INTERMEDIARY_MAPPINGS_NAME} mappings are created by {@link MapIntermediaryPlugin} tasks.
+ */
+public abstract class MapV2Plugin extends DefaultTaskedMappingsProjectPlugin<MapV2Plugin.Tasks> {
+    public static final String UNPICK_CLI_CONFIGURATION_NAME = UNPICK_NAME + "Cli";
+
+    @Override
+    protected Tasks applyImpl(@NotNull Project project) {
+        final Configuration unpickCli = project.getConfigurations().create(UNPICK_CLI_CONFIGURATION_NAME);
+
+        this.addDependencyWithCapability(
+            project.getDependencies(), unpickCli,
+            "quilt.internal:classpath-holders",
+            "quilt.internal:classpath-holders-unpick-cli"
+        );
+
+        // apply required plugins and save their registered objects
+        final PluginContainer plugins = project.getPlugins();
+
+        final QuiltMappingsExtension ext = plugins.apply(QuiltMappingsBasePlugin.class).getExt();
+
+        final MinecraftJarsPlugin.Tasks minecraftJarsTasks =
+            plugins.apply(MinecraftJarsPlugin.class).getTasks();
+        final TaskProvider<DownloadMinecraftLibrariesTask> downloadMinecraftLibraries =
+            minecraftJarsTasks.downloadMinecraftLibraries();
+
+        final MapMinecraftJarsPlugin.Tasks mapMinecraftJarsTasks =
+            plugins.apply(MapMinecraftJarsPlugin.class).getTasks();
+        final TaskProvider<MapPerVersionMappingsJarTask> mapPerVersionMappingsJar =
+            mapMinecraftJarsTasks.mapPerVersionMappingsJar();
+        final TaskProvider<InvertPerVersionMappingsTask> invertPerVersionMappings =
+            mapMinecraftJarsTasks.invertPerVersionMappings();
+        final TaskProvider<AddProposedMappingsTask> insertAutoGeneratedMappings =
+            mapMinecraftJarsTasks.insertAutoGeneratedMappings();
+
+        // register this plugin's tasks
+        final TaskContainer tasks = project.getTasks();
+
+        final var mergeTinyV2 = tasks.register(
+            MergeTinyV2Task.MERGE_TINY_V_2_TASK_NAME,
+            MergeTinyV2Task.class,
+            task -> {
+                // TODO QUESTION this used to be dependent on v2UnmergedMappingsJar, but afaict it has no effect on this task
```

Comment:
Please check this

Reply:
Yarn doesnt have `v2UnmergedMappingsJar` anymore, its probably fine